proposals/spline-animation/README.md

```diff
@@ -0,0 +1,1980 @@
+
+```c++
+bool UsdUtilsReduceSeriesInLayer(
+    const SdfLayerHandle &layer,
+    const TsReductionParams &params);
+
+bool UsdUtilsReduceSeriesOnStage(
+    const UsdStagePtr &stage,
+    const TsReductionParams &params,
+    const UsdEditTarget &editTarget = UsdEditTarget());
+```
+
+`ReduceSeriesOnStage` is potentially problematic, because it could lead to a
+mixture of reduced and unreduced opinions in various layers.  But it certainly
+does seem convenient.  If clients use a topmost layer as an edit target, that
+should suffice for reading.
+
+## Utilities
+
+The following general-purpose series and spline utilities will be available:
+
+```c++
+// Returns the bounding interval of regions over which the specified series
+// will evaluate to different values.  The returned interval may be infinite on
+// either side when extrapolation differs.  The returned interval is a bound,
+// and it is possible that there are regions within that bound where the series
+// do not differ.
+//
+GfInterval TsFindDifferingInterval(
+    const TsSeries &s1,
+    const TsSeries &s2);
+
+// Removes as many knots as possible from the specified intervals of a spline
+// without introducing error greater than maxErrorFraction * value range in
+// intervals.  Extremes (high or low points) are always preserved, and they are
+// detected by finding knots that are above or below their neighbors by at least
+// maxExtremeFraction * value range in intervals.
+//
+void TsSimplifySpline(
+    TsSpline *spline,
+    const GfMultiInterval &intervals = GfMultiInterval::GetFullInterval(),
+    double maxErrorFraction = .01,
+    double maxExtremeFraction = .001);
+
+// Finds a new set of knots that describe a similar spline, but with possibly
+// fewer knots.  First densely samples the spline by adding splits quantized
+// to samplingInterval (both absolutely and in spacing).  Then calls
+// TsSimplifySpline on the result.  Operates only within specified intervals.
+//
+void TsResampleSpline(
+    TsSpline *spline,
+    double samplingInterval,
+    const GfMultiInterval &intervals = GfMultiInterval::GetFullInterval(),
+    double maxErrorFraction = .01,
+    double maxExtremeFraction = .001);
+
+```
+
+# QUESTIONS
+
+Feedback is welcome on the following questions (or any other aspect of this
+proposal).
+
+## Splines of vectors?
+
+See [Value Type Categories](#value-type-categories).
+
+Should USD Anim support splines of floating-point vectors?  Any type that
+supports addition and scalar multiplication can be represented by a spline.
+Presto supports splines of floating-point vectors, but we have barely used them.
+
+Among other things, values and tangents of vector-valued splines can't easily be
+visualized in a 2D interface, making them less useful for artists.  These kinds
+of splines, if they are useful at all, would presumably be used for programmatic
+interpolation.
+
+Our guess for now is **no**, we should not support splines of vectors in USD
+Anim.
```

Comment:
Referring back to my message at the top level, a use case for this in a more generic curve data is for Color ramps in shaders.  These are well visualised and defined in many DCC's already for the described interpolation types.